src/main/java/com/connexta/search/index/IndexServiceImpl.java

```diff
@@ -7,70 +7,81 @@
 package com.connexta.search.index;
 
 import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
 
-import com.connexta.search.common.SearchManager;
+import com.connexta.search.common.Index;
+import com.connexta.search.common.IndexRepository;
 import com.connexta.search.common.exceptions.SearchException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URI;
 import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Pattern;
-import javax.validation.constraints.Size;
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.core.io.Resource;
+import org.apache.tika.exception.TikaException;
 import org.springframework.http.HttpStatus;
-import org.springframework.web.reactive.function.client.WebClient;
-import reactor.core.publisher.Mono;
 
 @Slf4j
 @AllArgsConstructor
 public class IndexServiceImpl implements IndexService {
 
-  @NotNull private final SearchManager searchManager;
-  @NotNull private final WebClient webClient;
+  @NotNull private final IndexRepository indexRepository;
+  @NotNull private final IonResourceLoader ionResourceLoader;
+  @NotNull private final BodyContentExtractor bodyContentExtractor;
 
   @Override
-  public void index(
-      @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 32, max = 32) final String datasetId,
-      @NotNull final URI irmUri) {
-    final Resource resource;
+  public void index(Index index) {
```

Comment:
This list of parameter will not grow. It is not the same as everything that gets indexed. I still think that the controller shouldn't be aware at all about how things are indexed. If it should pass an object instead of a few params, I don't think that object should be the Index object or even an IndexBuilder. The IndexService should decide what gets indexed, not the Controller.